rpmlint/checks/FileDigestCheck.py

```diff
@@ -9,16 +9,135 @@
+#
+# - shell like files (configuration files, shell scripts, Python
+# scripts, Perl scripts etc) that can contain empty lines or comments
+# introduced by '#'.
+# - XML files for things like D-Bus configuration files or polkit policies.
+# Here also XML style comments and whitespace can occur.
+#
+# The filter classes help in calculating a file digest for a filtered version
+# of the target file that is normalized and therefore stable against the
+# outlined kinds of changes. Since this makes getting the right whitelisting
+# digests hard, a small companion tool in tools/get_whitelisting_digest.py
+# exists.
+
+
+class DefaultDigester:
+    """This class performs the default digest calculation of arbitrary file
+    contents as is."""
+
+    def __init__(self, path, halg):
+        self.path = path
+        self.halg = halg
+
+    def parse_content(self):
+        # simply generate chunks on binary level
+        with open(self.path, 'rb') as fd:
+            chunk = fd.read(4096)
+            if not chunk:
+                return
+            yield chunk
+
+    def get_digest(self):
+        """This returns the hash digest of the *un*filtered input file."""
+        hasher = hashlib.new(self.halg)
+
+        for chunk in self.parse_content():
+            hasher.update(chunk)
+
+        return hasher.hexdigest()
+
+
+class ShellDigester(DefaultDigester):
+    """This class performs digest calculation of shell style configuration
+    files or scripts. Certain aspects like comments and whitespace will be
+    filtered out of the digest calculation."""
+
+    def parse_content(self):
+        # generate filtered lines
+        with open(self.path) as fd:
+            for line_nr, line in enumerate(fd):
+                line_nr += 1
+
+                stripped = line.strip()
+                if not stripped:
+                    # skip empty lines
+                    continue
+                elif line_nr == 1 and stripped.startswith('#!'):
+                    # keep shebang lines instact
+                    pass
+                elif stripped.startswith('#'):
+                    # skip comments
+                    # NOTE: we don't strip trailing comments like in
+                    # 'if [ 5 -eq $NUM ]; then # compare to 5'
+                    # because the danger would be too high to remove actual
+                    # content instead of just comments
+                    continue
+
+                # don't use the completely stripped version, in Python for
+                # example the indentation is part of the syntax and changes
+                # here could change the meaning
+                yield (line.rstrip() + '\n').encode()
+
+
+class XmlDigester(DefaultDigester):
+    """This class performs digest calculation of XML configuration files.
+    Certain aspects like comments and whitespace will be filtered out of the
+    digest calculation."""
+
+    def parse_content(self):
+        import xml.etree.ElementTree as ET
```

Comment:
Please import it in module context, this seems to me like not necessary optimization.